Taskfile.yml

```diff
@@ -66,6 +67,12 @@ tasks:
         vars:
           CHECKSUM_FILE: "{{.CHECKSUM_FILE}}"
           DATA_DIR: "{{.OUTPUT_DIR}}"
+          EXCLUDE_PATHS: &emsdk_checksum_exclude_paths
```

Comment:
When I diff emsdk before and after compilation, I get this:

```
upstream/emscripten/__pycache__: emcc.cpython-310.pyc
upstream/emscripten/cache: sanity.txt
upstream/emscripten/cache: symbol_lists
upstream/emscripten/cache: symbol_lists.lock
upstream/emscripten/cache/sysroot/lib/wasm32-emscripten: lto
upstream/emscripten/third_party/leb128/__pycache__: __init__.cpython-310.pyc
upstream/emscripten/tools/__pycache__: __init__.cpython-310.pyc
upstream/emscripten/tools/__pycache__: building.cpython-310.pyc
upstream/emscripten/tools/__pycache__: cache.cpython-310.pyc
upstream/emscripten/tools/__pycache__: colored_logger.cpython-310.pyc
upstream/emscripten/tools/__pycache__: config.cpython-310.pyc
upstream/emscripten/tools/__pycache__: diagnostics.cpython-310.pyc
upstream/emscripten/tools/__pycache__: emscripten.cpython-310.pyc
upstream/emscripten/tools/__pycache__: extract_metadata.cpython-310.pyc
upstream/emscripten/tools/__pycache__: feature_matrix.cpython-310.pyc
upstream/emscripten/tools/__pycache__: filelock.cpython-310.pyc
upstream/emscripten/tools/__pycache__: js_manipulation.cpython-310.pyc
upstream/emscripten/tools/__pycache__: line_endings.cpython-310.pyc
upstream/emscripten/tools/__pycache__: link.cpython-310.pyc
upstream/emscripten/tools/__pycache__: minimal_runtime_shell.cpython-310.pyc
upstream/emscripten/tools/__pycache__: response_file.cpython-310.pyc
upstream/emscripten/tools/__pycache__: settings.cpython-310.pyc
upstream/emscripten/tools/__pycache__: shared.cpython-310.pyc
upstream/emscripten/tools/__pycache__: system_libs.cpython-310.pyc
upstream/emscripten/tools/__pycache__: tempfiles.cpython-310.pyc
upstream/emscripten/tools/__pycache__: toolchain_profiler.cpython-310.pyc
upstream/emscripten/tools/__pycache__: utils.cpython-310.pyc
upstream/emscripten/tools/__pycache__: webassembly.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: __init__.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: boost_headers.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: bullet.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: bzip2.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: cocos2d.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: freetype.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: giflib.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: harfbuzz.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: icu.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: libjpeg.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: libmodplug.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: libpng.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: mpg123.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: ogg.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: regal.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2_gfx.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2_image.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2_mixer.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2_net.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2_ttf.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sdl2.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: sqlite3.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: vorbis.cpython-310.pyc
upstream/emscripten/tools/ports/__pycache__: zlib.cpython-310.pyc
upstream/emscripten/tools/ports/contrib/__pycache__: glfw3.cpython-310.pyc
```

I guess we should add some of those directories?

I also realized that maps are only supported in Taskfile variables starting from v3.37, so it's probably good to specify `Task >= v3.37` in the README if possible.

Reply:
> I guess we should add some of those directories?

Right, some of those files do not necessarily change after initial setup but it makes sense to add those to the exclusion list. Below are now also excluded:
```
            - "upstream/emscripten/__pycache__"
            - "upstream/emscripten/third_party/leb128/__pycache__"
            - "upstream/emscripten/tools/ports/__pycache__"
            - "upstream/emscripten/tools/ports/contrib/__pycache__"
```

> I also realized that maps are only supported in Taskfile variables starting from v3.37, so it's probably good to specify Task >= v3.37 in the README if possible.

True. In fact, according to their [changelogs](https://taskfile.dev/changelog/#v3380---2024-06-30) the feature isn't available as non-experimental until 3.38. I have updated the README to include the information.

Reply:
Good point.

Reply:
I think we can drop the explanation since it's really something we should add to the checksumming utils docs. So just `* [Task] 3.38.0 or higher` should be fine.